ui/lib/pki/addon/components/pki-role-form.js

```diff
@@ -0,0 +1,72 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+
+/**
+ * @module PkiRoleForm
+ * PkiRoleForm components are used to create and update PKI roles.
+ *
+ * @example
+ * ```js
+ * <PkiRoleForm @model={{this.model}}/>
+ * ```
+ * @callback onCancel
+ * @callback onSave
+ * @param {Object} model - Pki-role-engine model.
+ * @param {function} onChange - Handle the input coming from the custom yield field.
+ * @param {onCancel} onCancel - Callback triggered when cancel button is clicked.
+ * @param {onSave} onSave - Callback triggered on save success.
+ */
+
+export default class PkiRoleForm extends Component {
+  @service store;
+  @service flashMessages;
+
+  @tracked errorBanner;
+  @tracked invalidFormAlert;
+  @tracked modelValidations;
+  @tracked notAfter;
+  @tracked ttl;
+
+  @task
+  *save(event) {
+    event.preventDefault();
+    try {
+      const { isValid, state, invalidFormMessage } = this.args.model.validate();
+      this.modelValidations = isValid ? null : state;
+      this.invalidFormAlert = invalidFormMessage;
+      if (isValid) {
+        const { isNew, name } = this.args.model;
+        this.args.model.notAfter = this.notAfter;
```

Comment:
I'm wondering if this will create problems down the line having these attrs not being defined on the model, it may prevent them from returning from the backend properly when editing

Reply:
yeah, I ended up changing this flow based on saving the attrs to the model in the new component.